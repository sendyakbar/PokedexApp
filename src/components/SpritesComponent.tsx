import React, {memo} from 'react';
import {
  StyleSheet,
  View,
  Text,
  Image,
  ViewStyle,
  TextStyle,
  ImageStyle,
  Dimensions,
} from 'react-native';

const {width: DEVICE_WIDTH}: {width: number} = Dimensions.get('window');

type Props = {
  label: string;
  pics: string[];
};

const SpritesComponent = memo((props: Props): JSX.Element => {
  const {label, pics} = props;

  return (
    <>
      <Text style={styles.labelText}>{label}:</Text>
      <View style={styles.picsContainer}>
        {pics.map((item: string, index: number) => (
          <Image source={{uri: item}} style={styles.img} key={index} />
        ))}
      </View>
    </>
  );
});

type StyleType = {
  picsContainer: ViewStyle;
  img: ImageStyle;
  labelText: TextStyle;
};

const styles = StyleSheet.create<StyleType>({
  picsContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  img: {
    height: DEVICE_WIDTH / 4 - 8,
    width: DEVICE_WIDTH / 4 - 8,
  },
  labelText: {
    fontSize: 14,
    color: 'black',
    fontWeight: '700',
  },
});

export default SpritesComponent;
