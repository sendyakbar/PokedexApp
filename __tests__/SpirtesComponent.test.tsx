import 'react-native';
import React from 'react';
import {it} from '@jest/globals';
import renderer from 'react-test-renderer';

import SpritesComponent from '../src/components/SpritesComponent';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <SpritesComponent
        label="Test"
        pics={[
          'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/2.png',
          'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/2.png',
        ]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
