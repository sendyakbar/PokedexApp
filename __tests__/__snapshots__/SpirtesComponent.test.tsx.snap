// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <Text
    style={
      {
        "color": "black",
        "fontSize": 14,
        "fontWeight": "700",
      }
    }
  >
    Test
    :
  </Text>,
  <View
    style={
      {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/2.png",
        }
      }
      style={
        {
          "height": 179.5,
          "width": 179.5,
        }
      }
    />
    <Image
      source={
        {
          "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/2.png",
        }
      }
      style={
        {
          "height": 179.5,
          "width": 179.5,
        }
      }
    />
  </View>,
]
`;
